packages/x-archetype-utils/src/i18n/i18n.plugin.ts

```diff
@@ -0,0 +1,154 @@
+import { VueConstructor } from 'vue';
+import VueI18n, { LocaleMessageObject } from 'vue-i18n';
+import { deepMerge } from '@empathyco/x-deep-merge';
+import { AnyMessages, Device, I18nOptions, LoadLazyMessagesByDevice, Locale } from './i18n.types';
+
+/**
+ * I18n settings manager.
+ */
+export class I18n {
```

Comment:
With the types that we modified, we are losing message types. Errors that previously were triggered  now they aren't. We can solve them if we add a generic parameter to this class representing the full messages object for a given lang, and we use it across the methods or properties.